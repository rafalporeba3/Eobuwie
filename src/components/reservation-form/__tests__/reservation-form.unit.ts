import { shallowMount, Wrapper }            from '@vue/test-utils';
import { reservationFormConfigurationMock } from '@/components/reservation-form/__tests__/reservation-form-mock';
import cPrice                               from '@/components/price/price.component.vue';
import cReservationForm                     from '../reservation-form.component.vue';

describe('Reservation form', () => {
  let wrapper!: Wrapper<any>;

  beforeEach(() => {
    wrapper = shallowMount(cReservationForm, {
      propsData: { reservationFormConfiguration: reservationFormConfigurationMock },
    });
  });

  it('should render cReservationForm component with proper class', () => {
    const reservationFormComponent: Wrapper<any> = wrapper.findComponent(cReservationForm);

    expect(reservationFormComponent.exists())
      .toBeTruthy();

    expect(reservationFormComponent.classes())
      .toContain('c-reservation-form');
  });

  it('should contain cPrice component', () => {
    const priceComponent: Wrapper<any> = wrapper.findComponent(cPrice);

    expect(priceComponent.exists())
      .toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(wrapper.html())
      .toMatchSnapshot();
  });
});
