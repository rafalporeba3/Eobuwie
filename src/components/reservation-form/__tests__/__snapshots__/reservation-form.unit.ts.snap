// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reservation form should match snapshot 1`] = `
<div class="c-reservation-form">
  <c-price-stub price="100"></c-price-stub>
</div>
`;
